methods.py

```diff
@@ -1147,26 +1171,28 @@ def format_key_value(v):
         sys.path.remove(tmppath)
         sys.modules.pop("msvs")
 
+    extensions = {}
+    extensions["headers"] = [".h", ".hh", ".hpp"]
+    extensions["sources"] = [".c", ".cc", ".cpp"]
+    extensions["others"] = [".mm", ".inc", ".natvis", ".glsl", ".rc"]
```

Comment:
We also have `.m` files for iOS, I don't know if it's relevant to add.

And `.java` files for Android, which I'm told can also be a target for Visual Studio, though I don't know if we support this.

Reply:
I added the ones that I noted were either being added manually in the old logic, or filetypes that are specific to windows. It's useful to have text files in the solution so that search works even if they're not used for anything. `.a` files are library archives and won't do much for us in this situation, I think (but for generating other types of IDE projects, probably useful).